doc/user/content/sql/drop-index.md

```diff
@@ -16,6 +16,9 @@ do not have any indexes.)
 Field | Use
 ------|-----
 _view&lowbar;name_ | The name of the index you want to remove.
+`CASCADE` | Automatically removes any objects that depend on the index, as well as the index.
+`IF EXISTS`  |  Do not issue an error if the named index doesn't exist.
+`RESTRICT`  |  Remove the index, converting the materialized view to a non-materialized view. _(Default)._
```

Comment:
This is the wrong description here; the other table has the correct description of `RESTRICT`.

Reply:
@sploiselle @JLDLaughlin This is what I get when I try it:

```
SHOW FULL VIEWS;
     name      | type | materialized 
---------------+------+--------------
 t2_view       | user | t
 t_view        | user | t
```

`DROP INDEX t_view_primary_idx;`

```
SHOW FULL VIEWS;
     name      | type | materialized 
---------------+------+--------------
 t2_view       | user | t
 t_view        | user | f
```

Is this not converting the materialized view to a nonmaterialized view instead of removing it?

Reply:
This comment was on the `RESTRICT` keyword, but the example you're providing elides it from the command. If you re-run the command using `RESTRICT` does it still drop the index?

Reply:
@msudberg You're right––it should be implied. This looks like a bug.

Reply:
Ah, ok––it looks like we shouldn't have `RESTRICT` (or `CASCADE`) as an option on indexes yet because we don't have a notion of "dependency" for indexes afaict. The materialized view does not depend on the index (instead, the relationship goes the other way: the index depends on the view, though dropping the view doesn't require an explicit `CASCADE` to drop the index for ergonomic reasons). Something that would depend on the index would be something like a constraint, e.g. `NOT NULL`.

For future reference, a better place to pull the supported syntax from would be one of the test drive test files; in this case `test/testdrive/createdrop.td`. You can find the appropriate file by looking for the command you want (usually in all caps) in `*.td` files using e.g. VSCode (I think you use Atom, but there should be a similar search filtering option there).

One could make an argument that we shouldn't accept either cascade or restrict options because they're both definitely no-ops and it's confusing as to what to expect, but one could also make an argument that this is working as intended and obeys both the requested `CASCADE` and `RESTRICT` behavior.

@JLDLaughlin Does this track for you, as well?

Reply:
Yes, it does! Thank you both for digging into this. 

@sploiselle - updating materialized views to depend on indexes would... I'm not sure about all the ramifications, but create a cyclical dependency that would probably just require a lot of special handling. Do you think this is worth pursuing? Or do you think the current `RESTRICT` and `CASCADE` behavior (however weird) is sufficient?

Reply:
@sploiselle - I actually did search using Atom! `DROP INDEX CASCADE` shows up in materialize/test/sqllogictest/cockroach/alter_table.slt, but I don't see anything relevant in my local version of testdrive.td.

@JLDLaughlin - Is it actually correct to say that `DROP INDEX` converts materialized views to non-materialized views, or should the definition just be the usual "Removes materialized views"?

Reply:
@msudberg Ah––there are, of course, a million contingencies. A lot of features in the `materialize/test/sqllogictest/cockroach/` directory are not supported. The set of "canonical" tests is in `materialize/test/sqllogictest`.

I realize you asked this to Jessica, but maybe I can help get you unblocked: `DROP INDEX` removes the index, which converts a materialized view into a non-materialized view iff there are no indices remaining after this index gets dropped. `DROP INDEX` doesn't/can't drop the view it's on, so it would be a little confusing to say it removes the materialized view because the view itself remains, just in unmaterialized form.

Reply:
@sploiselle, @JLDLaughlin - What is the final call on including or omitting `RESTRICT` and `CASCADE`?

Reply:
I would suggest mentioning that it's valid syntax but is a no-op because nothing in Materialize can depend on indices. I don't think there's a clear answer around whether to e.g. include it in the railroad diagram, so defer to your judgment on the presentation.

Reply:
I think if CASCADE, RESTRICT, and no switch all do the exact same thing, we might as well omit CASCADE and RESTRICT and prevent the confusion. We do have other DROP commands that don't support those switches.

Reply:
Helping prevent the confusion from a documentation perspective would be to explain the contingency where users might encounter it, rather than papering over it by omitting the keyword and hoping no one types it in. There is definitely engineering work we could do to improve this UX, but a note in the docs takes a much less significant amount of effort and is a good intermediary solution until the engineering team can take on more of this polish work.